include/exadg/postprocessor/solution_field.h

```diff
@@ -41,45 +41,61 @@ class SolutionField : public dealii::Subscriptor
   using VectorType = dealii::LinearAlgebra::distributed::Vector<Number>;
 
   SolutionField()
-    : recompute_solution_field([](VectorType &, VectorType const &) {}),
+    : initialize_vector([](std::shared_ptr<VectorType> &) {}),
+      recompute_solution_field([](VectorType &, VectorType const &) {}),
       type(SolutionFieldType::scalar),
       name("solution"),
       dof_handler(nullptr),
       is_available(false),
-      solution(nullptr)
+      src_vector(nullptr)
   {
   }
 
+  /**
+   * This function initializes the DoF vector, the main data object of this class.
+   * This is done by using the lambda initialize_vector. The vector solution_vector
+   * may also point to external data.
+   */
   void
-  reinit(VectorType const & solution_in)
+  reinit()
+  {
+    initialize_vector(solution_vector);
+  }
+
+  /**
+   * This function invalidates the solution vector.
+   */
+  void
+  invalidate()
   {
     is_available = false;
-    solution     = &solution_in;
   }
 
   void
-  evaluate()
+  evaluate(VectorType const & src)
   {
     if(!is_available)
     {
-      recompute_solution_field(vector, *solution);
+      recompute_solution_field(*solution_vector, src);
       is_available = true;
     }
   }
 
-  VectorType &
-  get_vector_reference()
+  VectorType const &
+  get() const
   {
-    return vector;
+    AssertThrow(is_available,
+                dealii::ExcMessage("You are trying to access a Vector that is not available."));
+
+    return *solution_vector;
   }
 
   VectorType const &
-  get_vector() const
+  evaluate_get(VectorType const & src)
```

Comment:
I liked that evaluate_get() didn't need a src vector before. With this we had to be careful during initialization, but in the rest of the code we don't have to care about what's needed if we want the evaluated quantity.
Now (to get the evaluated quantity) it might be necessary to call e.g.
```
qunatity.evaluate_get(other_qunatity.evaluate_get(third_quantity))
```
instead of 
```
evaluate_get(quantity)
```

Reply:
I think this is the main point where we disagree. I like the first option, as it is clear what the input or src-vector is. The second option allows some misuse (wanted or unwanted) that I would like to avoid.

Reply:
I'm not sure how such a misuse could look like. Do you have an example?

In my opinion, it is more safe to provide the correct src-vector within the lambda at only one point in the code. 
On the other hand: If the user has to specify the src-vector every time he wants to access the field, it is more likely that an unwanted error is introduced. This might also lead to code bloat if the user wants to access the field in an arbitrary function, since this function also needs the src-vector(s) as input.

Reply:
I am referring to versions of the lambda function where the src parameter remained empty and some other variable of the PostProcessor class has been used instead. In my opinion it is safer to not operate on member variables but on the src, dst vectors passed to a function. I also believe it is easier to read/understand the code. I probably have a strong procedural perspective and believe that this is still a recommendable way of thinking for many problems we are dealing with in the context of PDE solvers. Its fine what we do not have identical perspectives here.

Reply:
What I had in mind was to remove the src parameter from the lambda completely and capturing the required src-vector in the lambda definition as it was done, e.g. for the vorticity_magnitude. This way, there is no need for a member variable in the SolutionField class. 
I can relate with both ways, but still prefer the version without handing over the src-vector every time.

Reply:
I think capturing the src-vector is difficult since the src-vector is not a member variable of `PostProcessor` in most cases and, therefore, not available during setup/initialization.

Reply:
True. I think it's OK for me the second way :)